python/taichi/aot/module.py

```diff
@@ -142,44 +142,46 @@ def add_kernel(self, kernel_fn, template_args=None, name=None):
         injected_args = []
         template_types = (NdarrayType, template)
         num_template_args = len([
-            anno for anno in kernel.argument_annotations
-            if isinstance(anno, template_types)
+            args.annotation for args in kernel.arguments
+            if isinstance(args.annotation, template_types)
         ])
         if template_args is not None and num_template_args != len(
                 template_args):
             raise TaichiCompilationError(
                 f'Need {num_template_args} inputs to instantiate the template '
                 f'parameters, got {len(template_args)}')
         i = 0
-        for arg_name, anno in zip(kernel.argument_names,
-                                  kernel.argument_annotations):
-            if isinstance(anno, template_types):
+        for args in kernel.arguments:
+            if isinstance(args.annotation, template_types):
                 if template_args:
-                    injected_args.append(template_args[arg_name])
+                    injected_args.append(template_args[args.name])
                 else:
-                    if not isinstance(anno, NdarrayType):
```

Comment:
Nit: if you'd like save typing a few words, you can create a new variable: `anno = arg.annotation`, and use `anno` below :-)

Reply:
yeah, it's a good suggestion and I will take it!